lib/Macros/Sources/ObservationMacros/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+        COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+        DEPENDS ${DEPENDENCY}
+        )
+      target_sources(${TARGET} PRIVATE
+        ${CMAKE_CURRENT_BINARY_DIR}/forced-${DEPENDENCY}-dep.swift
+      )
+    endforeach()
+  endfunction()
+
+  set(SWIFT_SYNTAX_MODULES
+    SwiftSyntax
+    SwiftSyntaxMacros
+  )
+
+  # Compute the list of SwiftSyntax targets
+  list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND "SwiftSyntax::"
+       OUTPUT_VARIABLE SWIFT_SYNTAX_TARGETS)
+
+  # TODO: Change to target_link_libraries when cmake is fixed
+  force_target_link_libraries(ObservationMacros PUBLIC
+    ${SWIFT_SYNTAX_TARGETS}
+  )
+
+  set(SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR
+      "${SWIFT_PATH_TO_EARLYSWIFTSYNTAX_BUILD_DIR}/lib/swift/host")
+  set(SWIFT_SYNTAX_LIBRARIES_DEST_DIR
+      "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/swift/host")
+
+  # Determine the SwiftSyntax shared library files that were built as
+  # part of earlyswiftsyntax.
+  list(TRANSFORM SWIFT_SYNTAX_MODULES PREPEND ${CMAKE_SHARED_LIBRARY_PREFIX}
+       OUTPUT_VARIABLE SWIFT_SYNTAX_SHARED_LIBRARIES)
+  list(TRANSFORM SWIFT_SYNTAX_SHARED_LIBRARIES APPEND
+       ${CMAKE_SHARED_LIBRARY_SUFFIX}
+       OUTPUT_VARIABLE SWIFT_SYNTAX_SHARED_LIBRARIES)
+
+  # Copy over all of the shared libraries from earlyswiftsyntax so they can
+  # be found via RPATH.
+  foreach (sharedlib ${SWIFT_SYNTAX_SHARED_LIBRARIES})
+    add_custom_command(
+      TARGET ObservationMacros PRE_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR}/${sharedlib} ${SWIFT_SYNTAX_LIBRARIES_DEST_DIR}/${sharedlib}
+      COMMENT "Copying ${sharedlib}"
+    )
+  endforeach()
+
+  # Copy all of the Swift modules from earlyswiftsyntax so they can be found
+  # in the same relative place within the build directory as in the final
+  # toolchain.
+  list(TRANSFORM SWIFT_SYNTAX_MODULES APPEND ".swiftmodule"
+       OUTPUT_VARIABLE SWIFT_SYNTAX_MODULE_DIRS)
+  foreach(module_dir ${SWIFT_SYNTAX_MODULE_DIRS})
+    file(GLOB module_files
+         "${SWIFT_SYNTAX_LIBRARIES_SOURCE_DIR}/${module_dir}/*.swiftinterface")
+    add_custom_command(
+      TARGET ObservationMacros PRE_BUILD
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${SWIFT_SYNTAX_LIBRARIES_DEST_DIR}/${module_dir}
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${module_files} ${SWIFT_SYNTAX_LIBRARIES_DEST_DIR}/${module_dir}/
+      COMMENT "Copying ${module_dir}"
+    )
+  endforeach()
+
+  target_include_directories(ObservationMacros PUBLIC
+    ${SWIFT_SYNTAX_LIBRARIES_DEST_DIR})
+
+  # Ensure the install directory exists before everything gets started
+  add_custom_command(
+      TARGET ObservationMacros PRE_BUILD
+      COMMAND ${CMAKE_COMMAND} -E make_directory "lib${LLVM_LIBDIR_SUFFIX}/host/plugins"
+  )
+
+  swift_install_in_component(TARGETS ObservationMacros
+    LIBRARY
+      DESTINATION "lib${LLVM_LIBDIR_SUFFIX}/host/plugins"
```

Comment:
This path seems like it's missing a `swift`?

Reply:
whoops, that is definitely missing - will follow up on that